Userland/Libraries/LibGfx/QOILoader.cpp

```diff
@@ -33,7 +34,7 @@ static ErrorOr<QOIHeader> decode_qoi_header(InputMemoryStream& stream)
     stream >> Bytes { &header, sizeof(header) };
     if (stream.handle_any_error())
         return Error::from_string_literal("Invalid QOI image: end of stream while reading header"sv);
-    if (StringView { header.magic, array_size(header.magic) } != QOI_MAGIC)
+    if (!AK::TypedTransfer<u8>::compare(header.magic, QOI_MAGIC, QOI_MAGIC_SIZE))
```

Comment:
What is being fixed here? The commit message doesn't present any argument for why this change is useful or good.

Also what's up with the `[Ugly]` prefix, please drop that and refrain from trying to add new semantics to commit messages without reaching prior concensus with the community.

Reply:
I tagged it as ugly, because I found it ugly, and I need to rethink the solution, or if anyone have a better solution.
The thing is that since in the current code consider the signature as a `StringView` it brings it's problems, which are tried to be resolved by this patch. Even if all the bytes are ASCII compatible a magic are just mere bytes, so using `StringView` here is problematic. This is the reason I used `TypedTransfer` here.

Reply:
`StringView` and `String` don't really promise any specific encoding, but if you're _that_ concerned, why not use `ReadonlyBytes` and its `operator==`?

TypedTransfer is a bit too low-level to use in such a high-level library imo (do note that `Span::operator==` uses this under the hood anyway).

Anyhow, I'm not sure what you find ugly about this (I haven't looked yet), but if you think it's ugly, then why would anyone want to look at it or merge it?